CI: Gateway API Conformance test: TestConformance/HTTPRouteListenerHostnameMatching failure

## CI failure

Gateway API Conformance test, `TestConformance`, `HTTPRouteListenerHostnameMatching`:
Some requests in `TestConformance/HTTPRouteListenerHostnameMatching` failed.

- [Link to CI test](https://github.com/cilium/cilium/actions/runs/4353915342/jobs/7608565752)
- [Raw logs](https://github.com/cilium/cilium/files/10909996/output.txt)
- [Related PR](https://github.com/cilium/cilium/pull/24215)

Output (extract):

```
=== NAME  TestConformance
    apply.go:175: Deleting tls-validity-checks-certificate 
    apply.go:175: Deleting certificate 
    apply.go:219: Deleting web-backend Deployment
    apply.go:219: Deleting web-backend Service
    apply.go:219: Deleting gateway-conformance-web-backend Namespace
    apply.go:219: Deleting app-backend-v2 Deployment
    apply.go:219: Deleting app-backend-v2 Service
    apply.go:219: Deleting app-backend-v1 Deployment
    apply.go:219: Deleting app-backend-v1 Service
    apply.go:219: Deleting gateway-conformance-app-backend Namespace
    apply.go:219: Deleting infra-backend-v3 Deployment
    apply.go:219: Deleting infra-backend-v3 Service
    apply.go:219: Deleting infra-backend-v2 Deployment
    apply.go:219: Deleting infra-backend-v2 Service
    apply.go:219: Deleting infra-backend-v1 Deployment
    apply.go:219: Deleting infra-backend-v1 Service
    apply.go:219: Deleting backend-namespaces Gateway
    apply.go:219: Deleting all-namespaces Gateway
    apply.go:219: Deleting same-namespace Gateway
    apply.go:219: Deleting gateway-conformance-infra Namespace
--- FAIL: TestConformance (117.08s)
    [...]
    --- FAIL: TestConformance/HTTPRouteListenerHostnameMatching (1.12s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/7_request_to_'no.matching.host/'_should_receive_a_404 (0.01s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/6_request_to_'foo.com/'_should_receive_a_404 (0.00s)
        --- FAIL: TestConformance/HTTPRouteListenerHostnameMatching/0_request_to_'bar.com/'_should_go_to_infra-backend-v1 (30.00s)
        --- FAIL: TestConformance/HTTPRouteListenerHostnameMatching/5_request_to_'multiple.prefixes.foo.com/'_should_go_to_infra-backend-v3 (30.00s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/4_request_to_'multiple.prefixes.bar.com/'_should_go_to_infra-backend-v3 (11.02s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/3_request_to_'boo.bar.com/'_should_go_to_infra-backend-v3 (11.02s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/2_request_to_'baz.bar.com/'_should_go_to_infra-backend-v3 (0.01s)
        --- PASS: TestConformance/HTTPRouteListenerHostnameMatching/1_request_to_'foo.bar.com/'_should_go_to_infra-backend-v2 (0.01s)
    [...]
/Simple_HTTP_request_should_reach_infra-backend (1.01s)
FAIL
FAIL	github.com/cilium/cilium/operator/pkg/gateway-api	117.132s
FAIL
##[error]Process completed with exit code 1.
```

--- a/.github/workflows/conformance-gateway-api.yaml
+++ b/.github/workflows/conformance-gateway-api.yaml
@@ -170,8 +170,7 @@ jobs:
             --gateway-class cilium \
             --supported-features ReferenceGrant,HTTPRoute,TLSRoute,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,RouteDestinationPortMatching,GatewayClassObservedGenerationBump,HTTPResponseHeaderModification \
             -test.run "TestConformance" \
-            -test.skip "TestConformance/HTTPRouteListenerHostnameMatching|TestConformance/HTTPRouteRedirectHostAndStatus" 
-            # Enable once HTTPRouteListenerHostnameMatchings #24217 is fixed
+            -test.skip "TestConformance/HTTPRouteRedirectHostAndStatus" 
             # Enable HTTPRouteRedirectHostAndStatus once https://github.com/kubernetes-sigs/gateway-api/issues/1805 is fixed upstream
 
       - name: Post-test information gathering

